core/src/test/scala/unit/kafka/tools/ConsoleProducerTest.scala

```diff
@@ -123,4 +163,37 @@ class ConsoleProducerTest {
     assertEquals("console-producer",
       producerConfig.getString(ProducerConfig.CLIENT_ID_CONFIG))
   }
+
+  @Test
+  def testBatchSizeOverride(): Unit = {
+    val config = new ConsoleProducer.ProducerConfig(batchSizeOverride)
+    val producerConfig = new ProducerConfig(ConsoleProducer.producerProps(config))
+    assertEquals("456",
+      producerConfig.getString(ProducerConfig.BATCH_SIZE_CONFIG))
```

Comment:
Test failed with 
```
ava.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.String (java.lang.Integer and java.lang.String are in module java.base of loader 'bootstrap')
```
It should be using getInt, and expected value should also be int 456, not string "456", like this:
`assertEquals(456, producerConfig.getInt(ProducerConfig.BATCH_SIZE_CONFIG))`

Same comments to below tests.